tests/components/sensor/test_version.py

```diff
@@ -0,0 +1,57 @@
+"""The test for the version sensor platform."""
+import unittest
+
+from homeassistant.setup import setup_component
+
+from tests.common import get_test_home_assistant
+
+MOCK_VERSION = '10.0'
+MOCK_DEV_VERSION = '10.0.dev0'
+
+
+class TestVersionSensor(unittest.TestCase):
+    """Test the Version sensor."""
+
+    def setup_method(self, method):
+        """Set up things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+
+    def teardown_method(self, method):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    def test_version_sensor(self):
+        """Test the Version sensor."""
+        config = {
+            'sensor': {
+                'platform': 'version',
+                'name': 'test',
+            }
+        }
+
+        assert setup_component(self.hass, 'sensor', config)
+
+        self.hass.states.set('sensor.test', MOCK_VERSION)
+        self.hass.block_till_done()
+
+        state = self.hass.states.get('sensor.test')
+
+        self.assertEqual(state.state, '10.0')
```

Comment:
Except for testing the setup of the sensor, the tests do not really test the function of the sensor but the state machine. Setting a state and getting a state like this can be done without this sensor setup.